what-is-pair.org

```diff
@@ -0,0 +1,251 @@
+  and a namespace called 'fun', then all services i expose from that namespace will
+  have a url of ~{SERVICENAME}.fun.cool.pair.sharing.io~
+- Prometheus and Grafana ::
+  These are dashboards for observing the health and activity of the cluster,
+  viewable at ~prometheus.cool.pair.sharing.io~ and
+  ~grafana.cool.pair.sharing.io~ (continuing the example username).
+- Knative(Serving) support ::
+  [[https://knative.dev][Knative]] is a method for running "serverless" components, or basically a solution for
+  easily deploying microservices.  A cluster can be configured to support Knative Serving
+  and Knative Events. Ours is setup for Serving
+- An in-cluster terminal/coding environment ::
+  This is [[https://github.com/sharingio/environment][sharingio/environment]], a containerized environment with a long set of
+  tools. It will be covered in more detail later in the doc as it contains much
+  of the 'pairing' part of pair.
+** Services chosen by the user
+Anytime a user requests a new cluster, they have the option to set the following, specific to that cluster:
+- Guests ::
+  Github users that you want to have access to the cluster. For any github
+  username added as a guest, Pair will fetch the user's github public keys and
+  add them to the pair Environment. They also adjust the Environment's git
+  template to include a `Co-Authored-By: Github User Name <github user email>`
+  line. This allows for the person to be able to ssh into the Pair Environment.
+- Repos ::
+  Github repositories that are cloned into the environment container upon its
+  startup. Multiple repos can be requested when creating a new cluster.
+** Services chosen by the project
+  If a github repository contains a .sharing.io folder with an init script, then
+  pair will run that script after cloning the repo into the Environment container.
+  The init scripts often include deploying project-specific services and infra to the cluster.
+
+  For example, if you request a pair instance loaded with ~cncf/apisnoop~, then
+  your instance will come with an Environemnt, along with an apisnoop namespace
+  where snoopdb is already running, and already loaded with the latest test logs
+  from kubernetes.
+
+  It could be possible to utilize the init script with the DNS of Pair to quickly deploy
+  an app with a public URL--e.g. a repo that builds and deploys a chat app available at
+  ~app.chat.cool.pair.sharing.io~
+
+** The Pair site
+Pair can also refer to the site at pair.sharing.io, which gives a limited view
+into the status and capabilities of your pair instances. From the site you can
+create a new instance via a web form, delete an instance, view the status for
+any of your instances, or download the necessary files and scripts for accessing
+an instance.
+
+Any services deployed and exposed with a url will be listed on the status page to
+quickly access them.  The status page also contains a link to the web view of the
```

Comment:
this includes Knative services

Reply:
resolved with [aae2998](https://github.com/sharingio/org/pull/1/commits/aae2998a37af4b534f29fe38b8786638d1c6d983)